libraries/appbase/include/appbase/application.hpp

```diff
@@ -65,14 +65,21 @@ class application_factory
       }
    }
 
-   application& new_application( const string& id);
+   application& new_application( const string& id );
+
+   application& new_application( const string& id, application app );
+
+   application& get_application( const string& id ){
+      return apps.at(id);
+   }
 
    template<typename Plugin>
    std::shared_ptr< abstract_plugin > new_plugin(){
       return new_plugin(Plugin::name());
    }
 
    std::shared_ptr< abstract_plugin > new_plugin(string name){
+      assert(plugin_factories.count(name));
```

Comment:
just a note: find might be used instead of count.

Reply:
If there is something I consider really unnecessary and ugly design pattern in std containers due to crap design, it is the construct 
```(a.find(key) != a.end()) ```

Fortunately, at least map() offers better way to check existence of an element :)